src/segmented-control/__tests__/desktop.test.tsx

```diff
@@ -115,6 +116,295 @@ describe('Segment disabled property', () => {
     getSegmentWrapper(segmentedControlWrapper, 0).getElement().click();
     expect(onChange).toHaveBeenCalledWith(expect.objectContaining({ detail: { selectedId: 'seg-1' } }));
   });
+
+  describe('Disabled with reason', () => {
```

Comment:
What about the case of using `disabledReason` without `disabled` set to `true`? Although this is not a "valid" use case, we want to have an established behavior for it (which to my understanding is the same as not passing any `disabledReason`) which shouldn't change, so it might be a good idea to cover this case as well.

And maybe we also want to emit a dev warning in such case?

Reply:
I see that the PR was updated but this in particular wasn't addressed

Reply:
I introduced this test case: https://github.com/cloudscape-design/components/pull/2414/commits/ee0e639c21aeff83e5395125342cd387ab41737b#diff-7d2444d48fe7dd36eb8a4e838395cff5269219a361dca51ea45a9e5c572c02dbR121

Reply:
Oh right, my bad